test_util/cluster.py

```diff
@@ -92,7 +97,7 @@ def from_hosts(
         num_agents,
         num_public_agents,
     ):
-        assert all(x > 0 for x in [num_masters, num_agents, num_public_agents]), (
+        assert all(x >= 0 for x in [num_masters, num_agents, num_public_agents]), (
```

Comment:
It was, and that's how I discovered this faulty assertion. But once I fixed this, I discovered my own assertion in `run_integration_tests()` that at least one public agent is required for tests. I added a public agent but figured I'd leave this fix here.